### ISSUE TITLE
TypeError: coercing to Unicode: need string or buffer, NoneType found - Debian 8.2 & Debian 7 Raspi

### ISSUE THREAD

**Issue by beatkind** [certbot/certbot#2131]
Getting some error with generating some certs. 

Logfile:

2016-01-09 23:22:12,068:DEBUG:letsencrypt.cli:Root logging level set at 30
2016-01-09 23:22:12,069:INFO:letsencrypt.cli:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2016-01-09 23:22:12,070:DEBUG:letsencrypt.cli:letsencrypt version: 0.1.1
2016-01-09 23:22:12,070:DEBUG:letsencrypt.cli:Arguments: ['--rsa-key-size', '4096', '-d', '5space.de', '-d', 'www.5space.de']
2016-01-09 23:22:12,071:DEBUG:letsencrypt.cli:Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#webroot,PluginEntryPoint#null,PluginEntryPoint#manual,PluginEntryPoint#standalone)
2016-01-09 23:22:12,076:DEBUG:letsencrypt.cli:Requested authenticator None and installer None
2016-01-09 23:22:14,841:DEBUG:letsencrypt.plugins.disco:Other error:(PluginEntryPoint#webroot): --webroot-path must be set
Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/plugins/disco.py", line 103, in prepare
    self._initialized.prepare()
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/plugins/webroot.py", line 53, in prepare
    self.option_name("path")))
PluginError: --webroot-path must be set
2016-01-09 23:22:14,845:DEBUG:letsencrypt.display.ops:Multiple candidate plugins: \* apache
Description: Apache Web Server - Alpha
Interfaces: IAuthenticator, IInstaller, IPlugin
Entry point: apache = letsencrypt_apache.configurator:ApacheConfigurator
Initialized: <letsencrypt_apache.configurator.ApacheConfigurator object at 0x7f7dcdfa8f10>
Prep: True
- standalone
  Description: Automatically use a temporary webserver
  Interfaces: IAuthenticator, IPlugin
  Entry point: standalone = letsencrypt.plugins.standalone:Authenticator
  Initialized: <letsencrypt.plugins.standalone.Authenticator object at 0x7f7dcd0c8c50>
  Prep: True
  2016-01-09 23:22:16,185:DEBUG:letsencrypt.cli:Selected authenticator <letsencrypt.plugins.standalone.Authenticator object at 0x7f7dcd0c8c50> and installer None
  2016-01-09 23:22:16,217:DEBUG:letsencrypt.cli:Picked account: <Account(decb0c4ee2f23b5f447fe57d8ebcefe9)>
  2016-01-09 23:22:16,219:DEBUG:root:Sending GET request to https://acme-v01.api.letsencrypt.org/directory. args: (), kwargs: {}
  2016-01-09 23:22:16,226:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
  2016-01-09 23:22:16,780:DEBUG:requests.packages.urllib3.connectionpool:"GET /directory HTTP/1.1" 200 263
  2016-01-09 23:22:16,784:DEBUG:root:Received <Response [200]>. Headers: {'Content-Length': '263', 'Expires': 'Sat, 09 Jan 2016 23:22:17 GMT', 'Strict-Transport-Security': 'max-age=604800', 'Server': 'nginx', 'Connection': 'keep-alive', 'Pragma': 'no-cache', 'Cache-Control': 'max-age=0, no-cache, no-store', 'Date': 'Sat, 09 Jan 2016 23:22:17 GMT', 'X-Frame-Options': 'DENY', 'Content-Type': 'application/json', 'Replay-Nonce': 'KMezolIY77wORhHGkNQantH_PXwGbNRC-INBhMwAxVg'}. Content: '{"new-authz":"https://acme-v01.api.letsencrypt.org/acme/new-authz","new-cert":"https://acme-v01.api.letsencrypt.org/acme/new-cert","new-reg":"https://acme-v01.api.letsencrypt.org/acme/new-reg","revoke-cert":"https://acme-v01.api.letsencrypt.org/acme/revoke-cert"}'
  2016-01-09 23:22:16,785:DEBUG:acme.client:Received response <Response [200]> (headers: {'Content-Length': '263', 'Expires': 'Sat, 09 Jan 2016 23:22:17 GMT', 'Strict-Transport-Security': 'max-age=604800', 'Server': 'nginx', 'Connection': 'keep-alive', 'Pragma': 'no-cache', 'Cache-Control': 'max-age=0, no-cache, no-store', 'Date': 'Sat, 09 Jan 2016 23:22:17 GMT', 'X-Frame-Options': 'DENY', 'Content-Type': 'application/json', 'Replay-Nonce': 'KMezolIY77wORhHGkNQantH_PXwGbNRC-INBhMwAxVg'}): '{"new-authz":"https://acme-v01.api.letsencrypt.org/acme/new-authz","new-cert":"https://acme-v01.api.letsencrypt.org/acme/new-cert","new-reg":"https://acme-v01.api.letsencrypt.org/acme/new-reg","revoke-cert":"https://acme-v01.api.letsencrypt.org/acme/revoke-cert"}'
  2016-01-09 23:22:16,790:DEBUG:letsencrypt.storage:Expected symlink /etc/letsencrypt/live/www.5space.de/cert.pem for cert does not exist.
  2016-01-09 23:22:16,791:DEBUG:letsencrypt.cli:Exiting abnormally:
  Traceback (most recent call last):
  File "/root/.local/share/letsencrypt/bin/letsencrypt", line 11, in <module>
    sys.exit(main())
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/cli.py", line 1396, in main
    return args.func(args, config, plugins)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/cli.py", line 598, in obtain_cert
    _auth_from_domains(le_client, config, domains)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/cli.py", line 379, in _auth_from_domains
    action, lineage = _treat_as_renewal(config, domains)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/cli.py", line 235, in _treat_as_renewal
    ident_names_cert, subset_names_cert = _find_duplicative_certs(config, domains)
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/cli.py", line 204, in _find_duplicative_certs
    candidate_names = set(candidate_lineage.names())
  File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/storage.py", line 459, in names
    with open(target) as f:
  TypeError: coercing to Unicode: need string or buffer, NoneType found

**Comment by penn5** [comment]
I am getting this error too on raspi debian gnu/linux 7

**Comment by beatkind** [comment]
getting it also with apache, standalone and webroot

**Comment by bmw** [comment]
I believe this is a duplicate of #1195. @beatkind, @penn5, did you delete, move, or modify files in `/etc/letsencrypt`?

**Comment by beatkind** [comment]
@bmw i haven't changed anything. Saw that issue also but doesn't work for me.

**Comment by bmw** [comment]
@beatkind, well I can tell you why the error occurred based on your log. There are files in `/etc/letsencrypt/renewal` that contain information about certificates `letsencrypt` previously obtained. On your system, one of these files (which would have been named `www.5space.de.conf` by default), contains a line like the following at the beginning of the file:
```
cert = /etc/letsencrypt/live/www.5space.de/cert.pem
```
This line tells `letsencrypt` that `/etc/letsencypt/live/www.5space.de/cert.pem` is a symlink to the current certificate for a specified set of domains. `letsencrypt` is reading this line and trying to access symlink at the specified path. The problem is this symlink doesn't exist and the client crashes.

The interesting thing is that the `www.5space.de.conf` configuration file is created by `letsencrypt` after the symlink at `/etc/letsencrypt/live/www.5space.de/cert.pem`. Do you have any idea how `/etc/letsencrypt/renewal/www.5space.de.conf` could have been modified or the symlink at `/etc/letsencrypt/live/www.5space.de/cert.pem` deleted? If not, can you describe to me what you have done `letsencrypt` so far?

**Comment by beatkind** [comment]
i tried to create some certs but it gots me errors back like can not check domain and some like that.

I tried it on another host system to create some certs with that domain an now it says there are to many request for 5space.de so there have to be created earlier. Seams like there where created when I played with lets encrypt and getting the error back with the authentic... Can we solve the problem about that there are to many requests? Because I don't have any certs... Don't now if you get the problem :s my english is not the best. @bmw
